src/main/java/com/hierynomus/msdtyp/SID.java

```diff
@@ -16,13 +16,45 @@
 package com.hierynomus.msdtyp;
 
 import com.hierynomus.protocol.commons.ByteArrayUtils;
+import com.hierynomus.protocol.commons.EnumWithValue;
 import com.hierynomus.protocol.commons.buffer.Buffer;
 import com.hierynomus.smbj.common.SMBBuffer;
 
 /**
  * [MS-DTYP].pdf 2.4.2 SecurityIdentifier SID
  */
 public class SID {
+
+    public enum SidType implements EnumWithValue<SidType> {
+        SID_TYPE_NONE(0, "0"),
+        SID_TYPE_USER(1, "User"),
+        SID_TYPE_DOM_GRP(2, "Domain group"),
```

Comment:
Rename this to `SID_TYPE_DOMAIN_GROUP` (Unless the current name is the actual name used in the documentation)